src/btree/bt_cursor.c

```diff
@@ -778,6 +832,53 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
     return (ret);
 }
 
+/*
+ * __btcur_search_near_bounds_reposition --
+ *     Search near with a bounded cursor should reposition the cursor at the nearest bound.
+ */
+static int
+__btcur_search_near_bounds_reposition(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
+{
+    WT_BTREE *btree;
+    WT_CURSOR *cursor;
+    bool upper_out_of_bounds, lower_out_of_bounds;
+
+    btree = CUR2BT(cbt);
+    cursor = &cbt->iface;
+    lower_out_of_bounds = upper_out_of_bounds;
+
+    WT_ASSERT(session, WT_CURSOR_HAS_BOUNDS(cursor));
+
+    if (F_ISSET(cursor, WT_CURSTD_BOUND_LOWER)) {
+        if (btree->type == BTREE_ROW)
+            /*
+             * We need to assume that the bounds are inclusive here, suppose a caller calls with the
```

Comment:
Just to confirm, we want to force inclusive here, so that we prevent the cursor from setting the raw key again? Is this because set_key is a costly operation?

Reply:
We force inclusive here so we don't re-set the key on the cursor yes. Is it costly? That would depend on the workload, but i imagine calling search_near on your lower or upper bound is more common than calling it with a key outside of the key range.